pkg/sink/sink.go

```diff
@@ -167,35 +167,39 @@ func createResources(resources []json.RawMessage, restClient restclient.Interfac
 
 // createResource uses the kubeClient to create the resource defined in the
 // TriggerResourceTemplate and returns any errors with this process
-func createResource(rt json.RawMessage, restClient restclient.Interface, discoveryClient discoveryclient.DiscoveryInterface, eventListenerNamespace string, eventListenerName string, eventID string, logger *zap.SugaredLogger) error {
+func (r Resource) createResource(rt json.RawMessage, triggerName string, eventID string) error {
 	// Assume the TriggerResourceTemplate is valid (it has an apiVersion and Kind)
 	apiVersion := gjson.GetBytes(rt, "apiVersion").String()
 	kind := gjson.GetBytes(rt, "kind").String()
 	namespace := gjson.GetBytes(rt, "metadata.namespace").String()
 	// Default the resource creation to the EventListenerNamespace if not found in the resource template
 	if namespace == "" {
-		namespace = eventListenerNamespace
+		namespace = r.EventListenerNamespace
 	}
-	apiResource, err := findAPIResource(discoveryClient, apiVersion, kind)
+	apiResource, err := findAPIResource(r.DiscoveryClient, apiVersion, kind)
 	if err != nil {
 		return err
 	}
 
-	rt, err = sjson.SetBytes(rt, "metadata.labels."+triggersv1.LabelEscape+triggersv1.EventListenerLabelKey, eventListenerName)
+	rt, err = sjson.SetBytes(rt, "metadata.labels."+triggersv1.LabelEscape+triggersv1.EventListenerLabelKey, r.EventListenerName)
```

Comment:
Tiny nit: Might be worth extracting the three calls to add labels to its own helper function for easy testing

Reply:
Done!